src/controller/CHIPDeviceController.cpp

```diff
@@ -945,6 +961,53 @@ void DeviceCommissioner::OnSessionEstablished()
     RendezvousCleanup(CHIP_NO_ERROR);
 }
 
+CHIP_ERROR DeviceCommissioner::SendOperationalCertificateSigningRequestCommand(NodeId remoteDeviceId)
+{
+    // TODO: Call OperationalCredentials cluster API to send command
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR DeviceCommissioner::OnOperationalCertificateSigningRequest(NodeId node, const uint8_t * csrBuf, uint32_t csrLength)
+{
+    ReturnErrorCodeIf(mOperationalCredentialsDelegate == nullptr, CHIP_ERROR_INCORRECT_STATE);
+
+    uint8_t opCert[kMaxCHIPOpCertLength];
```

Comment:
how large is this? are we sure an embedded device will have sufficient stack space to allocate for opcert and signcert data below? Consider unique_ptr and memoryalloc these

Reply:
I'll change these to memory allocs.
This code block should run on commissioners only. Do we anticipate embedded commissioner devices?